### ISSUE TITLE
[Feature Request] Custom Resolutions

### ISSUE THREAD

**Issue by WeeJeWel** [tml1024/FluffyDisplay#5]
Thank you for this great piece of software!

I would like to be able to enter a custom resolution (width x height x ppi).

**Comment by Cueball** [comment]
Yeah it's missing quite a few historically common resolutions; adding to or editing the list of resolutions would be invaluable.

I was looking for 1680x1050 (WSXGA+) - this was the resolution for 20" Cinema Display and 20" iMac (mid 2007 aka iMac7,1).

**Comment by tml1024** [comment]
> I was looking for 1680x1050 (WSXGA+) - this was the resolution for 20" Cinema Display and 20" iMac (mid 2007 aka iMac7,1).

Adding that is so trivial that I feel I can just ask you to submit a patch;)

**Comment by Cueball** [comment]
Patch done: https://github.com/tml1024/FluffyDisplay/pull/7 Hadn't looked to see how easy it was, so thanks for the tip :-)

But that doesn't eliminate the need to either add a custom resolution, or ability to edit the list of resolutions. Creating a PR, waiting for merge and release - sure, it gets the job done eventually, but efficient? :-/

Suggestion: maybe if there was something like a resolutions.txt file that was read on startup? Put the file in App Support; create a default file if it doesn't exist or isn't a correctly parseable list of resolutions. [Instead of hard-coding the list of resolutions to support, the default list is bundled as an asset and copied to App Support if there's a problem reading the file there.]

**Comment by tml1024** [comment]
I will create a new release within some days. Not sure when I will have time and interest to add custom resolutions. Would it be enough if one could add custom resolutions using a configuration file or something like that? Handling that would be easier than coding a UI just for it.

**Comment by Cueball** [comment]
> add custom resolutions using a configuration file 

Sure, that's what I was trying to propose (obviously not well!) with the "resolutions.txt" (or "resolutions.conf" or whatever) idea. Any user can simply define new resolutions by editing an existing text-based config file. [If the file doesn't exist, a default config file (included in the app bundle) is copied to the expected location. This should also happen if the config file is unreadable for some reason.]

No additional UI required, except maybe a UI option to "Reset config file to default values". Most (if not all) people wanting to use this app aren't going to be shy of editing a config file.

**Comment by tml1024** [comment]
Ah sorry, I had forgotten that you indeed proposed such a config file. Will implement this perhaps next weekend.

**Comment by mmm117** [comment]
It would be better if end user can enter custom resolution directly. Thanks!!
